float32 velocity
float32 angle
wall_dist hug
bool enabled
